app/helpers/routerHelper.js

```diff
@@ -1,19 +1,28 @@
 import { ErrorContext } from 'buying-catalogue-library';
-import { appBaseUri } from '../config';
+import { appBaseUri, isDevelopment } from '../config';
 import { getEndpoint } from '../endpoints';
+import { logger } from '../logger';
 
 export const withCatch = (authProvider, route) => async (req, res, next) => {
   try {
     return await route(req, res, next);
   } catch (err) {
-    if (err.response && err.response.status === 401) {
-      req.headers.referer = `${appBaseUri}${req.originalUrl}`;
-      return authProvider.login()(req, res, next);
+    if (err.stack) {
+      logger.error(`Unexpected Error:\n${err.stack}`);
     }
+
     if (err instanceof ErrorContext) {
       return next(err);
     }
-    const defaultError = new ErrorContext({ status: 500 });
+
+    if (err.response && err.response.status === 401) {
+      req.headers.referer = `${appBaseUri}${req.originalUrl}`;
+      return authProvider.login()(req, res, next);
+    }
+
+    const stackTrace = isDevelopment() ? err.stack : undefined;
+    const defaultError = new ErrorContext({ status: 500, stackTrace });
+
     return next(defaultError);
```

Comment:
I took this from the order-form but moved the stack-trace block higher so it would always be logged out which is much nicer for development. I don't think there is any hard for logging out stack-traces in prod either but I can add a condition for that.